stacks-signer/src/lib.rs

```diff
@@ -68,3 +73,59 @@ pub trait Signer<T: SignerEventTrait>: Debug + Display {
         command: Option<RunLoopCommand>,
     );
 }
+
+/// A wrapper around the running signer type for the signer
+pub type RunningSigner<T> = libsigner::RunningSigner<SignerEventReceiver<T>, Vec<SignerResult>, T>;
+
+/// The wrapper for the runloop signer type
+type RunLoopSigner<S, T> =
+    libsigner::Signer<RunLoopCommand, Vec<SignerResult>, RunLoop<S, T>, SignerEventReceiver<T>, T>;
+
+/// The spawned signer
+pub struct SpawnedSigner<S: Signer<T> + Send, T: SignerEventTrait> {
+    /// The underlying running signer thread handle
+    running_signer: RunningSigner<T>,
+    /// The command sender for interacting with the running signer
+    pub cmd_send: Sender<RunLoopCommand>,
+    /// The result receiver for interacting with the running signer
+    pub res_recv: Receiver<Vec<SignerResult>>,
+    /// Phantom data for the signer type
+    _phantom: std::marker::PhantomData<S>,
+}
+
+impl<S: Signer<T> + Send, T: SignerEventTrait> SpawnedSigner<S, T> {
+    /// Stop the signer thread and return the final state
+    pub fn stop(self) -> Option<Vec<SignerResult>> {
+        self.running_signer.stop()
+    }
+
+    /// Wait for the signer to terminate, and get the final state. WARNING: This will hang forever if the event receiver stop signal was never sent/no error occurred.
+    pub fn join(self) -> Option<Vec<SignerResult>> {
+        self.running_signer.join()
+    }
+}
+
+impl<S: Signer<T> + Send + 'static, T: SignerEventTrait + 'static> SpawnedSigner<S, T> {
+    /// Create a new spawned signer
+    pub fn new(config: GlobalConfig) -> Self {
+        let endpoint = config.endpoint;
+        info!("Starting signer with config: {:?}", config);
```

Comment:
This leaks private info in logs - use `.config_to_log_string()`

Reply:
Gonna just implement Debug manually to use config_to_log_string as would be easy to make this mistake elsewhere.